model-optimizer/extensions/front/tf/NonConstBeginStridedSliceReplacement.py

```diff
@@ -0,0 +1,136 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from extensions.ops.gather import Gather
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, rename_nodes
+from mo.ops.squeeze import Squeeze
+from mo.ops.unsqueeze import Unsqueeze
+
+
+class NonConstBeginStridedSliceReplacement(FrontReplacementSubgraph):
+    """
+    The transformation handles StridedSlice operation with dynamic begin and end values
+    when slicing performs along just one dimension with a dynamic index.
+    For example, StridedSlice with begin=(0,idx,0), end=(0,idx+1,0),
+    and begin_mask=end_mask=shrink_mask=(0, 1, 0) can be replaced with Squeeze(axis=1;Gather(axis=1; Unsqueeze(idx))).
+    The transformation attempts to match to following sub-graph:
+
+    Input ----> StridedSlice(begin_mask, end_mask, and shrink_mask where only element for axis equal to 1) --> OTHER OPS
+                   /\                  /\                  /\
+                   |                   |                   |
+      ---------> Pack(Begin)          Pack(End)         Const(Step) = (1,..,1)
+      |               /\                /\   /\
+      |               |                 |    |
+      |               |                 |   Const(All others)
+      |        Const(All others)        |
+    Index ---------------------------> Add
+                                        /\
+    Const(SliceSize)=1------------------|
+
+    And the original sub-graph is transformed as follows:
+
+    Input --------> Gather(axis) ---> Squeeze(axis) ---> OTHER OPS
+                       /\
+                       |
+    Index -----> Unsqueeze(axis=1)
+
+    """
+    enabled = True
+
+    @staticmethod
+    def pattern(**kwargs):
+        return dict(
+            nodes=[('begin', dict(op='Pack')),
+                   ('end', dict(op='Pack')),
+                   ('step', dict(op='Const')),
+                   ('strided_slice', dict(op='StridedSlice')),
+                   ],
+            edges=[('begin', 'strided_slice', {'in': 1}),
+                   ('end', 'strided_slice', {'in': 2}),
+                   ('step', 'strided_slice', {'in': 3})])
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        strided_slice_node = match['strided_slice']
+        begin_node = match['begin']
+        end_node = match['end']
+        step_node = match['step']
+
+        # retrieve attribute values
+        begin_mask = strided_slice_node.soft_get('begin_mask')
+        end_mask = strided_slice_node.soft_get('end_mask')
+        shrink_mask = strided_slice_node.soft_get('begin_mask')
```

Comment:
If `soft_get` is used to get attributes, then maybe default values are should be specified?

Reply:
I will add default value for `shrink_mask` since this is not mandatory attribute. I am leaning to use `soft_get` even for mandatory attributes since it is our API under which different data structures can be used.

Reply:
Done.